runtime/src/accounts_db.rs

```diff
@@ -6620,6 +6629,7 @@ impl AccountsDb {
         }
     }
 
+    // TODO: confirm this is only called from replay and fg processes.
```

Comment:
These todos seem probably true today. I agree we should confirm this. I imagine we'd find deadlocks if something about the code changes and we encounter thread pool crossing code at some point in the future.

Reply:
Besides a mainnet node which was running with this code for a while, I have not done any other form of testing yet to see if this deadlocks or not.

What was the setup that you were observing dead-locks?
Meaning how long did it take until you it dead-locked? and, was it just a normal validator running on mainnet, or was it in a test setup?

Reply:
It was likely a mnb validator. If it isn't deadlocking using these thread pools in your tests, then we most likely aren't aggravating that or something else has changed somewhere else in the system. I'm confident I was trying to use the clean thread pool.